File: src/web/src/render/effect/NoEffect.ts
import type { Effect } from './Effect.js';

export class NoEffect implements Effect {
  trigger(_x: number, _y: number): void {}
  isActive(): boolean { return false; }
  applyToPixels(_src: Uint8ClampedArray, _dst: Uint8ClampedArray, _w: number, _h: number): void {}
  stop(): void {}
}

File: src/main/java/com/ullgren/modern/simple3d/render/NoEffect.java
package com.ullgren.modern.simple3d.render;

/**
 * Null-object implementation of {@link Effect}: does nothing.
 * Used when the user selects "No Effect" from the Effect menu.
 */
public class NoEffect implements Effect {

  @Override
  public void trigger(int x, int y) {}

  @Override
  public boolean isActive() { return false; }

  @Override
  public void applyToPixels(int[] src, int[] dst, int w, int h) {}
}
